### PR TITLE
[FIX] #159. pos_restaurant and pos_required_customer are incompatible

### PR DESCRIPTION
**Pull request by legalsylvain**
Fix : #159.

Trivial fix to make pos_customer_required and pos_restaurant compatible.

(if pos_restaurant is installed, there is no pos order in `this.pos.get('selectedOrder')` when the PoS is starting. (only when a table is selected).

With this patch, (and if required setting is "order") first odoo will ask the place, and just after, the name of the customer.

@damendieta : could you test ?

CC reviewers of the first PR : @pedrobaeza, @invitu, @JosDeGraeve, @StefanRijnhart 

regards.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please change module version number and check Travis

**Comment by legalsylvain** [comment]
@pedrobaeza : done.

File: pos_customer_required/models/pos_config.py [hunk 1]
```diff
@@ -18,8 +18,8 @@ class PosConfig(models.Model):
     ]
 
     require_customer = fields.Selection(
-        selection=_REQUIRE_CUSTOMER_KEYS,
-        string='Require Customer',
+        selection=_REQUIRE_CUSTOMER_KEYS, required=True,
```
Comment (pedrobaeza) [inline pos_customer_required/models/pos_config.py]: Don't make it required. It's not needed.

Comment (legalsylvain) [reply pos_customer_required/models/pos_config.py]: Well, it was an error in fact. if it's not required 4 value are possible, and not 3. 
I don't understand what is the problem, if a default value is set. 
thanks for your answer.

Comment (yelizariev) [reply pos_customer_required/models/pos_config.py]: It may be either required False and two options in Selection, or required True and three options in selection. I'd prefer required=True

Comment (pedrobaeza) [reply pos_customer_required/models/pos_config.py]: Adding required fields on existing models makes tests of other modules that inherit from _point_of_sale_ to not work due to a not null constraint in the DB and the code that defaults it to the correct value not yet loaded. It's what it has happened with all the *_warn fields, and Odoo has removed it the `required` flag on them on master. I prefer to avoid this.

Comment (pedrobaeza) [reply pos_customer_required/models/pos_config.py]: FYI, I have just found the problem on one of l10n-spain modules and remove it: https://github.com/OCA/l10n-spain/commit/07f182134896976bac42fda92958c889030539bf

Comment (rafaelbn) [reply pos_customer_required/models/pos_config.py]: Please @legalsylvain could you check this last comment?

Comment (yelizariev) [reply pos_customer_required/models/pos_config.py]: @pedrobaeza is it still true when we have ``default='value'`` attribute, i.e. does odoo adds sql ``DEFAULT`` attribute in case of having static default value?

**Comment by rafaelbn** [comment]
Hi @legalsylvain , I trying to test this FIX but when I go to runbot

http://3324612-237-b2b1e9.runbot2.odoo-community.org/?db=3324612-237-b2b1e9-all

This redirects to https://runbot2.odoo-community.org/runbot and don't open the Odoo for this PR

Do you know why? :thinking:  and idea?

If possible please check travis because is :red_circle: after that I will test it and if it FIX https://github.com/OCA/pos/issues/159 we can merge ASAP :smiley:

File: pos_customer_required/models/pos_config.py (hunk 1 above)
Comment (yelizariev) [reply pos_customer_required/models/pos_config.py]: ok, it seems that odoo doesn't use ``DEFAULT`` on db level

**Comment by rafaelbn** [comment]
@legalsylvain travis says:

```
2018-01-11 01:44:37,987 5185 ERROR openerp_test openerp.addons.pos_customer_required.tests.test_pos: ERROR
2018-01-11 01:44:38,002 5185 INFO openerp_test openerp.addons.pos_customer_required.tests.test_pos: ======================================================================
2018-01-11 01:44:38,002 5185 ERROR openerp_test openerp.addons.pos_customer_required.tests.test_pos: ERROR: test_customer_not_required (openerp.addons.pos_customer_required.tests.test_pos.TestPosRequireCustomer)
2018-01-11 01:44:38,002 5185 ERROR openerp_test openerp.addons.pos_customer_required.tests.test_pos: Traceback (most recent call last):
2018-01-11 01:44:38,002 5185 ERROR openerp_test openerp.addons.pos_customer_required.tests.test_pos: `   File "/home/travis/build/OCA/pos/pos_customer_required/tests/test_pos.py", line 13, in test_customer_not_required
2018-01-11 01:44:38,002 5185 ERROR openerp_test openerp.addons.pos_customer_required.tests.test_pos: `     posconfig.require_customer = 'no'
2018-01-11 01:44:38,002 5185 ERROR openerp_test openerp.addons.pos_customer_required.tests.test_pos: `   File "/home/travis/odoo-9.0/openerp/fields.py", line 847, in __set__
2018-01-11 01:44:38,002 5185 ERROR openerp_test openerp.addons.pos_customer_required.tests.test_pos: `     value = self.convert_to_cache(value, record)
2018-01-11 01:44:38,002 5185 ERROR openerp_test openerp.addons.pos_customer_required.tests.test_pos: `   File "/home/travis/odoo-9.0/openerp/fields.py", line 1545, in convert_to_cache
2018-01-11 01:44:38,002 5185 ERROR openerp_test openerp.addons.pos_customer_required.tests.test_pos: `     raise ValueError("Wrong value for %s: %r" % (self, value))
2018-01-11 01:44:38,002 5185 ERROR openerp_test openerp.addons.pos_customer_required.tests.test_pos: ` ValueError: Wrong value for pos.config.require_customer: 'no'
2018-01-11 01:44:38,003 5185 INFO openerp_test openerp.addons.pos_customer_required.tests.test_pos: Ran 2 tests in 0.661s
2018-01-11 01:44:38,003 5185 ERROR openerp_test openerp.addons.pos_customer_required.tests.test_pos: FAILED
2018-01-11 01:44:38,003 5185 INFO openerp_test openerp.addons.pos_customer_required.tests.test_pos:  (errors=1)
2018-01-11 01:44:38,003 5185 ERROR openerp_test openerp.modules.module: Module pos_customer_required: 0 failures, 1 errors
2018-01-11 01:44:38,702 5185 INFO openerp_test openerp.modules.module: module pos_empty_home: creating or updating database tables
```

File: pos_customer_required/models/pos_config.py (hunk 1 above)
Comment (yelizariev) [reply pos_customer_required/models/pos_config.py]: I post the proposial https://github.com/odoo/odoo/issues/24461

**Comment by rafaelbn** [comment]
Hello @legalsylvain we can merge this quickly but travis is :red_circle: do you have any idea why?

File: pos_customer_required/models/pos_config.py [hunk 2]
```diff
@@ -12,15 +12,11 @@ class PosConfig(models.Model):
     _inherit = 'pos.config'
 
     _REQUIRE_CUSTOMER_KEYS = [
-        ('no', 'Optional'),
```
Comment (chienandalu) [inline pos_customer_required/models/pos_config.py:15]: The lack of this is breaking your tests

Comment (legalsylvain) [reply pos_customer_required/models/pos_config.py:15]: Hi. I have no time to fix that point, but feel free to make commit on that PR to fix that problem.

Comment (rafaelbn) [reply pos_customer_required/models/pos_config.py:15]: :smile: @legalsylvain we are not using POS in v9 :thinking: we also cannot take time here as we are working hard en v11 :+1:  we just wanted to help and have v9 branch green
